nornir/plugins/connections/netmiko.py

```diff
@@ -36,20 +36,25 @@ def open(
         connection_options: Optional[Dict[str, Any]] = None,
         configuration: Optional[Config] = None,
     ) -> None:
-        parameters = {
+
+        parameters: Dict[str, Union[str, int]] = {
             "host": hostname,
             "username": username,
             "password": password,
-            "port": ssh_port,
         }
+        if ssh_port:
+            parameters["port"] = ssh_port
 
         if nos is not None:
             # Look device_type up in corresponding map, if no entry return the host.nos unmodified
             device_type = napalm_to_netmiko_map.get(nos, nos)
             parameters["device_type"] = device_type
 
-        netmiko_connection_args = connection_options or {}
-        netmiko_connection_args.update(parameters)
+        if connection_options is None:
+            connection_options = {}
+
+        netmiko_connection_args = parameters
+        netmiko_connection_args.update(connection_options)
```

Comment:
Not critical, but something that crossed my mind:
we could name things in the connection plugin as plugin-agnostic as possible. For example:
`napalm_to_netmiko_map` -> `PLATFORM_MAPPING`
`netmiko_connection_args` -> `connection_args`
(naming is just an example, not saying it should be named this way)

The benefit would be that if a user wants to implement a custom plugin, he could take literally any existing plugin, do minimal changes and have a new connection plugin.
I don't want to force my ideology here, so I will leave the decision to you.